saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/nlp/BaseTypes/BaseTyps.md

```diff
@@ -0,0 +1,31 @@
+# Saul NLP Base Types
+Built in hierarchy of common constituents for processing text and NLP tasks
+and a relation class that connects those constituents.
+Nlp base types help to design data models in Saul and provide many built in feature 
+extractors and helpers.
+
+## Hierarchy
+All classes of the hierarchy assumed to have a unique **id** throughout the entire corpora,
+a **text**, and also character based span which shows the **start** index and exclusive 
+**end** index of each constituent.
+
+On the top level of the hierarchy we have the [`Document`](Document.java) class.
+Each document contains many [`Sentences`](Segment.java) which in turn can 
+have many [`Phrases`](Phrases) and finally each phrase have many [`Tokens`](Token.java). 
+Note that, you can omit one or more of these constituents for specific usages.
+
+### Properties
+We can specify additional properties for all hierarchy classes using
+`setPropertyValue`. This function adds a value to the list of values for that property. 
```

Comment:
Do you mean when there are XML tags related to a linguistic unit that we want to use as properties of that linguistic unit, right? if so, please make it clear as such.